gemcalibration/include/gem/calib/Calibration.h

```diff
@@ -54,15 +54,113 @@ namespace gem {
           throw (xgi::exception::Exception);
 
         std::vector<Calibration*> v_gemcal;
 
-        int PulseDelay;
+        calType_t m_calType;
 	
+        std::map<calType_t, std::map<std::string, uint32_t>> m_scanParams{
+            {PHASE  ,{{"nSamples",0},{"trigType", 0},}},
+            {LATENCY,{
+                {"nSamples"  , 100},
+                {"trigType"  , 0},
+                {"l1aTime"   , 250},
+                {"calPhase"  , 0},
+                {"mspl"      , 4},
+                {"scanMin"   , 0},
+                {"scanMax"   , 255},
+                {"vfatChMin" , 0},
+                {"vfatChMax" , 0},
+                {"vt2"       , 0},
+		{"throttle"  ,100},
+                {"signalSourceType"       , 0},
+                }},
+            {SCURVE,{
+                {"nSamples"  , 100},
+		{"trigType"  , 0}, // TODO: TTC local should be only possible one
+                {"l1aTime"   , 250},
+                {"pulseDelay", 40},
+                {"latency"   , 33},
+                {"vfatChMin" , 0},
+                {"vfatChMax" , 127},
+                {"calPhase"  , 0},
+                }},
+	    {SBITRATE  ,{
+		{"nSamples",0},
+		{"comparatorType",0},
+		{"perChannelType",0},
+		{"vfatChMin" , 0},
+                {"vfatChMax" , 0},
+		}},
+	    {THRESHOLD  ,{
+		{"nSamples"  ,0},
+		{"trigType"  , 0},
+		{"dataType"  , 0},
+		{"vfatChMin" , 0},
+		{"vfatChMax" , 0},
+		{"vt2"       , 0},
+		}},
+            {TRIMDAC  , {
+		{"nSamples"   , 0}, 
+		{"trigType"   , 0}, // TODO: TTC local should be only possible one
+		{"nSamples"   , 0},
+		{"l1aTime"    , 250},
+                {"pulseDelay" , 40},
+                {"latency"    , 33},
+		{"mspl"       , 4},
+		{"trimValues" , 0},// TODO: need to be implemented properly as taking array of numbers 
+                                   // TODO: need to implement interaction with DB to get proper configurations per ARM DAC
+		}},
+	    {DACSCANV3  ,{
+	       {"nSamples",0},
+	       {"adcType",0}
+	      }},// TODO: drop down with DACs to select to scan on, and a select all button
+	    {TEMPERATURE  ,{// TODO: Move to monitor page
+	       {"nSamples"      ,0},
+	       {"tempSensorType",0},
+	       {"timeInterval"  ,0},
+	      }},
+            {SBITREADOUT  ,{{"acquisitionTime",60},}},
+	    {SBITMAPANDRATE  ,{
+	       {"timeIntRate",1},
+	       {"rates", 1000},// TODO: need to be implemented properly as taking array of numbers 
+	       }},
+	    {CALIBRATEARMDAC,{
+	       {"nSamples"  , 100},
+	       {"trigType"  , 0}, // TODO: TTC local should be only possible one
+               {"l1aTime"   , 250},
+               {"pulseDelay", 40},
+               {"latency"   , 33},
+               {"armDacList" , 0},// TODO: need to be implemented properly as taking array of numbers 
+            
+               {"calPhase"  , 0}, 
+	       }}  
+        };
+	std::map< std::string,std::string > m_scanParamsLabels{
```

Comment:
I guess if we are going to introduce labels, we should think about merging all these maps into structures. 
Fine for the moment, but should be refactored later.